code/controllers/_DynamicAreaLighting_TG.dm

```diff
@@ -306,20 +264,27 @@
 		//show the dark overlay so areas, not yet in a lighting subarea, won't be bright as day and look silly.
 			SetLightLevel(4)
 
-#undef LIGHTING_LAYER
-#undef LIGHTING_CIRCULAR
-//#undef LIGHTING_ICON
-#define LIGHTING_MAX_LUMINOSITY_STATIC	8	//Maximum luminosity to reduce lag.
-#define LIGHTING_MAX_LUMINOSITY_MOBILE	5	//Moving objects have a lower max luminosity since these update more often. (lag reduction)
-#define LIGHTING_MAX_LUMINOSITY_TURF	1	//turfs have a severely shortened range to protect from inevitable floor-lighttile spam.
+#undef USE_CIRCULAR_LIGHTING
 
 //set the changed status of all lights which could have possibly lit this atom.
 //We don't need to worry about lights which lit us but moved away, since they will have change status set already
 //This proc can cause lots of lights to be updated. :(
 /atom/proc/UpdateAffectingLights()
-	for(var/atom/A in oview(LIGHTING_MAX_LUMINOSITY_STATIC-1,src))
-		if(A.light)
-			A.light.changed = 1			//force it to update at next process()
+
+/atom/movable/UpdateAffectingLights()
+	if(isturf(loc))
+		loc.UpdateAffectingLights()
+
+/turf/UpdateAffectingLights()
+	if(affecting_lights)
+		for(var/thing in affecting_lights)
+			thing:changed = 1			//force it to update at next process()

```

Comment:
Either typecast (without checking if you are concerned with speed) or at least rename the var into something describing what exactly is the things that is supposed the be updated in next process()
I know that affecting_lights is supposed to only hold special things, but its for better understandability of code